qiita_db/support_files/patches/python_patches/14.py

```diff
@@ -0,0 +1,20 @@
+# Feberuary 7, 2015
+# This patch recreates all the QIIME mapping files to avoid lower-upper case
+# problems. See https://github.com/biocore/qiita/issues/799
+
+from qiita_db.util import get_mountpoint
+from qiita_db.sql_connection import SQLConnectionHandler
+from qiita_db.metadata_template import PrepTemplate
+
+conn_handler = SQLConnectionHandler()
+
+_id, fp_base = get_mountpoint('templates')[0]
+
+for prep_template_id in conn_handler.execute_fetchall(
+        "SELECT prep_template_id FROM qiita.prep_template"):
+    prep_template_id = prep_template_id[0]
+    pt = PrepTemplate(prep_template_id)
+    study_id = pt.study_id
+
+    for _, fpt in pt.get_filepaths():
+        pt.create_qiime_mapping_file(fpt)
```

Comment:
Just to be clear, AFAIK the code only accepts 1 sample template per study but several prep templates. All sample and prep templates stay "forever" in the base_dir/templates/ folder. Once a prep template is processed a qiime mapping file is also generated from combining the sample and prep templates. I think the way to go will be to reprocess the file mapping files, simply because we can assure that both the raw files and the data in the db are the same. Note that you have to be careful cause if you drop and then add templates, you might be erasing other levels like process data, etc.